src/components/TablePagination/TablePagination.tsx

```diff
@@ -0,0 +1,115 @@
+import { FC, useEffect, useState } from "react";
+import Box from "@mui/material/Box";
+import Divider from "@mui/material/Divider";
+import FormControl from "@mui/material/FormControl";
+import MenuItem from "@mui/material/MenuItem";
+import Pagination from "@mui/material/Pagination";
+import Typography from "@mui/material/Typography";
+import Select, { SelectChangeEvent } from "@mui/material/Select";
+
+import { useUniversalSearchParams } from "../../hooks";
+import { itemsPerPageOptions } from "../../constants";
+import {
+  boldTextStyles,
+  defaultPaginationStyles,
+  itemsPageStyles,
+  itemsShownWrapper,
+  opacityTextStyles,
+  pageItemsWrapper,
+  paginationContainerStyles,
+  paginationDividerStyles,
+} from "./TablePaginationStyles";
+import { formControlStyles } from "../../theme";
+
+export interface ITablePaginationProps {
+  totalItems: number;
+  totalPages: number;
+}
+
+export const TablePagination: FC<ITablePaginationProps> = ({ totalItems, totalPages }) => {
+  const { searchParams, setParam } = useUniversalSearchParams();
+
+  const initialPage = Number(searchParams.get("page")) || 1;
```

Comment:
- [ ] create a custom hook for the pagination logic

Reply:
@AndriiKasian  Added custom hook for pagination logic